dnSpy/dnSpy/Documents/TreeView/DocumentTreeView.cs

```diff
@@ -753,16 +753,11 @@ sealed class AddDocumentInfo {
 			TreeNodeData newSelectedNode = null;
 			for (int i = 0, j = 0; i < filenames.Length; i++) {
 				// Resolve shortcuts
-				var directory = Path.GetDirectoryName(filenames[i]);
-				var file = Path.GetFileName(filenames[i]);
-
-				var shell = new Shell32.Shell();
-				var folder = shell.NameSpace(directory);
-				var folderItem = folder.ParseName(file);
-				if (folderItem != null && folderItem.IsLink) {
-					var link = (Shell32.ShellLinkObject)folderItem.GetLink;
-					filenames[i] = link.Path;
-				}
+				var ws = new IWshRuntimeLibrary.WshShell();
```

Comment:
This one could also be placed inside the try block.

Since exceptions are slow and a user can drag many files, you should set a boolean to false if an exception is thrown and not execute this code again.

Reply:
It can be a local initialized before the loop.